packages/expect/src/utils.ts

```diff
@@ -16,3 +20,25 @@ export function recordAsyncExpect(test: any, promise: Promise<any> | PromiseLike
 
   return promise
 }
+
+export function wrapSoft(utils: Chai.ChaiUtils, fn: (this: Chai.AssertionStatic & Assertion, ...args: any[]) => void) {
+  return function (this: Chai.AssertionStatic & Assertion, ...args: any[]) {
+    const test = utils.flag(this, 'vitest-test')
+
+    const state: MatcherState = test?.context._local
+      ? test.context.expect.getState()
+      : getState((globalThis as any)[GLOBAL_EXPECT])
+
+    if (!state.soft)
+      return fn.apply(this, args)
+
+    try {
+      return fn.apply(this, args)
+    }
+    catch (err) {
+      test.result.state = 'softFail'
```

Comment:
I'm not sure if we should introduce new states for this. I think `fail` is enough - otherwise it might break reporters